InnoGotchiBackend/InnnoGotchi.DAL/Respositories/RequestRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using InnnoGotchi.DAL.EF;
+using InnnoGotchi.DAL.Entities;
+using InnnoGotchi.DAL.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace InnnoGotchi.DAL.Respositories
+{
+    /// <summary>
+    /// Represents repository that has full access to coloboration request entities in database
+    /// </summary>
+    public class RequestRepository : IRepository<ColoborationRequest>
+    {
+        private InnoGotchiContext _context;
+        public RequestRepository(InnoGotchiContext context)
+        {
+            _context = context;
+        }
+
+        public bool Contains(Func<ColoborationRequest, bool> predicate)
+        {
+            ColoborationRequest? request = FirstOrDefault(predicate);
+            if (request == null)
+                return false;
+
+            return true;
+        }
+
+        public void Create(ColoborationRequest item)
+        {
+            _context.Requests.Add(item);
+        }
+
+        public bool Delete(int id)
+        {
+            ColoborationRequest? request = Get(id);
+            if (request != null)
+            {
+                _context.Requests.Remove(request);
+                return true;
+            }
+
+            return false;
+        }
+
+        public IQueryable<ColoborationRequest> FindAll(Func<ColoborationRequest, bool> predicate)
+        {
+            return GetAll().Where(predicate).AsQueryable();
+        }
+
+        public ColoborationRequest? FirstOrDefault(Func<ColoborationRequest, bool> predicate)
+        {
+            return GetAll().FirstOrDefault(predicate);
+        }
+
+        public ColoborationRequest? Get(int id)
+        {
+            ColoborationRequest? request = FirstOrDefault(r => r.Id == id);
+
+            if (request != null)
+            {
+                _context.Entry(request).State = EntityState.Detached;
+                return request;
+            }
+            else
+                return null;
+        }
+
+        public IQueryable<ColoborationRequest> GetAll()
+        {
+            return _context.Requests.Include(r => r.RequestOwner)
+                                    .Include(r => r.RequestReceipient);
+        }
+
+        public void Update(ColoborationRequest item)
+        {
+            _context.Entry(item).State = EntityState.Detached;
```

Comment:
Why we need this state???

Reply:
Without this state throwing this expception
![image](https://user-images.githubusercontent.com/74291930/216836450-0195f291-a1f5-488c-96b6-7e61acf45977.png)
Even such state don't help
![image](https://user-images.githubusercontent.com/74291930/216836570-8ccaabff-de3a-4b8a-805c-920674e7c5f4.png)

I was trying to resolve this problem by different ways but only this state helps